src/main/java/jetbrains/buildServer/auth/oauth/ServerPrincipalFactory.java

```diff
@@ -28,15 +28,15 @@ public ServerPrincipalFactory(@NotNull UserModel userModel) {
     }
 
     @NotNull
-    public ServerPrincipal getServerPrincipal(@NotNull final String userName, @NotNull Map<String, String> schemeProperties) {
+    public ServerPrincipal getServerPrincipal(@NotNull final String userId, @NotNull final String userName, final String userEmail, @NotNull Map<String, String> schemeProperties) {
         ServerPrincipal existingPrincipal = findExistingPrincipal(userName);
         if (existingPrincipal != null) {
             LOG.info("Use existing user: " + userName);
             return existingPrincipal;
         }
 
         boolean allowCreatingNewUsersByLogin = AuthModuleUtil.allowCreatingNewUsersByLogin(schemeProperties, DEFAULT_ALLOW_CREATING_NEW_USERS_BY_LOGIN);
-        Map<PropertyKey, String> userProperties = ImmutableMap.<PropertyKey, String>of(PluginConstants.ID_USER_PROPERTY_KEY, userName);
+        Map<PropertyKey, String> userProperties = ImmutableMap.<PropertyKey, String>of(PluginConstants.ID_USER_PROPERTY_KEY, userId);
```

Comment:
part of the issue I'm trying to fix here is that currently, using google oauth2 will cause the ID_USER_PROPERTY_KEY to be based off a user's name.  THis means any google account with the same full name will login to the same teamcity user.

Unfortunately, my solution is not backwards compatible, and I'm not sure it is possible to migrate.